recon/recon.py

```diff
@@ -68,6 +71,46 @@ def country_message(query, country, message=''):
     else:
         print(f"{Bcolors.WARNING} {query} could not determine location. Country extracted is None.{Bcolors.ENDC}\n")
 
+def scoper(assessment_id, file):
+    if file is None:
+        raise ValueError("File cannot be None")
+
+    file_path = file.name  # Extract filename
+    output_file = f"{assessment_id}_Scoper.txt"
+
+    try:
+        # Run nmap and capture output
+        nmap_result = subprocess.run(
+            ["nmap", "-Pn", "-n", "-sL", "-iL", file_path], 
+            stdout=subprocess.PIPE, 
+            text=True, 
+            check=True
+        ).stdout
+
+        # Process output with cut
+        cut_result = subprocess.run(
+            ["cut", "-d", " ", "-f", "5"],
+            input=nmap_result,  # Pass as string
+            stdout=subprocess.PIPE,
+            text=True,
+            check=True
+        ).stdout
+
+        # Remove unwanted lines using grep
+        grep_result = subprocess.run(
+            ["grep", "-v", "nmap\|address"],
+            input=cut_result,  # Pass as string
+            stdout=subprocess.PIPE,
+            text=True,
+            check=True
+        ).stdout
+
+        # Write final output to the file
+        with open(output_file, "w") as out_f:
+            out_f.write(grep_result)
+
```

Comment:
I would suggest adding some lines that print out to the user where the file is going to be saved so they know that it worked as expected.

Reply:
Script has been tested and everything looks good.